src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java

```diff
@@ -138,21 +141,17 @@ public List<Unit> unitAttack(String token, Position attacker, Position defender)
         }
         ensureWithinRange(attacker);
         ensureWithinRange(defender);
-        Optional<Unit> attackingUnitOptional = getUnitAt(attacker);
-        if (attackingUnitOptional.isEmpty())
-            throw new UnitNotFoundException(attacker);
-        Unit attackingUnit = attackingUnitOptional.get();
-        Optional<Unit> defendingUnitOptional = getUnitAt(defender);
-        if (defendingUnitOptional.isEmpty())
+        Unit attackingUnit = getUnitAtPosition(attacker);
+        if (attackingUnit == null)
             throw new UnitNotFoundException(attacker);
-        Unit defendingUnit = defendingUnitOptional.get();
+        Unit defendingUnit = getUnitAtPosition(defender);
+        if (defendingUnit == null)
+            throw new UnitNotFoundException(defender);
         if (this.decoratedPlayers.get(token).getId() != attackingUnit.getUserId())
             throw new WrongUnitOwnerException(attackingUnit, this.decoratedPlayers.get(token).getId());
         if (attackingUnit.getTeamId() == defendingUnit.getTeamId())
             throw new WrongTargetTeamException(attackingUnit, defendingUnit);
         attackingUnit.attack(defendingUnit);
-        // TODO: attacking does not move the unit (setPosition), but interface implies it does!
-        // logger.move(turnNumber);
         if (defendingUnit.getHealth() <= 0)
```

Comment:
We still need to remove the unit

Reply:
But we can also do another issue for that

Reply:
![WhatsApp Image 2022-05-12 at 08 53 51](https://user-images.githubusercontent.com/91155454/168009901-7e6cc877-c588-43b6-9549-6b8eb266ee72.jpeg)
At least from the PlayerDecorator it will be removed. And since we should be getting our units from the PlayerDecorator, I hope it works.